docs/content/en/docs/implementing/tasks/_index.md

```diff
@@ -5,12 +5,72 @@ weight: 90
 hidechildren: false # this flag hides all sub-pages in the sidebar-multicard.html
 ---
 
-Keptn tasks are defined in a
+A
 [KeptnTaskDefinition](../../yaml-crd-ref/taskdefinition.md/)
-resource.
+resource defines tasks that the Keptn Lifecycle Toolkit runs
+as part of the pre- and post-deployment phases of a
+[KeptnApp](../../yaml-crd-ref/app.md) or
+[KeptnWorkload](../../concepts/workloads.md).
+
+A Keptn task runs as an application
+[container](https://kubernetes.io/docs/concepts/containers/),
+which runs as part of a Kubernetes
+[job](https://kubernetes.io/docs/concepts/workloads/controllers/job/).
 A task definition includes a function
 that defines the action taken by that task.
-It can be configured in one of three different ways:
+
+To implement a Keptn task:
+
+- Define a
+  [KeptnTaskDefinition](../yaml-crd-ref/taskdefinition.md)
+  resource that defines the container
+- [Annotate your workloads](../integrate/#annotate-workloads)
+  to integrate your task with Kubernetes
+- Add your task to the  [KeptnApp](../yaml-crd-ref/app.md)
+  resource that associates your `KeptnTaskDefinition`
+  with the pre- and post-deployment tasks that should run in it
+
+This page provides information to help you create your tasks:
+
+- Code your task in a [container](#containers)
+- Understand how to use [Context](#context)
+  that contains a Kubernetes cluster, a user, a namespace,
+  the application name, workload name, and version.
+- Use [parameterized functions](#parameterized-functions)
+  if your task requires input parameters
+- [Create secret text](#create-secret-text)
+  and [pass secrets to a function](#pass-secrets-to-a-function)
+  if necessary.
+
+## Containers
+
+Each `KeptnTaskDefinition` can use exactly one container,
```

Comment:
```suggestion
Each `KeptnTaskDefinition` can use exactly one runner,
```

Reply:
But they are also limited to a single container, too, aren't they?  So what if this becomes "...can use exactly one container with one runner"?  It seems that we need to maintain the association with containers.  For `container-runtime`, is it possible to provide an image that uses more than one runner?

Reply:
Yes, you can select only one runner between deno, python, and container. Then for each runner you can have:

- deno: you can run a single script
- python: you can run a single script
- container: you can run a single container

Behind the scene, KLT handles all three runners as K8s Job. Hence, behind the scene, each KeptnTask is mapped to a single Job which spawns a single Pod that contains a single container.

Reply:
Did the "one container with one runtime" thing